src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -293,6 +298,50 @@ private function autowireMethod(\ReflectionMethod $reflectionMethod, array $argu
     }
 
     /**
+     * Autowires getters.
+     *
+     * @param array $overridenGetters
+     * @param array $autowiredMethod
+     *
+     * @return array
+     */
+    private function autowireOverridenGetters(array $overridenGetters, array $autowiredMethod)
+    {
+        foreach ($autowiredMethod as $reflectionMethod) {
+            if (isset($overridenGetters[strtolower($reflectionMethod->name)])
+                || !method_exists($reflectionMethod, 'getReturnType')
+                || 0 !== $reflectionMethod->getNumberOfParameters()
+                || $reflectionMethod->isFinal()
+                || $reflectionMethod->returnsReference()
+                || !($returnType = $reflectionMethod->getReturnType())
+            ) {
+                continue;
+            }
+
+            if (null === $this->types) {
```

Comment:
This logic must be updated to account for https://github.com/symfony/symfony/pull/21494/files#diff-62df969ae028c559d33ffd256de1ac49R248 before this line